tools/dep_updaters/update-zlib.sh

```diff
@@ -1,23 +1,47 @@
 #!/bin/sh
 set -e
-# Shell script to update zlib in the source tree to a specific version
+# Shell script to update zlib in the source tree to the most recent version.
+# Zlib rarely creates tags or releases, so we use the latest commit on the main branch.
+# See: https://github.com/nodejs/node/pull/47417
 
 BASE_DIR=$(cd "$(dirname "$0")/../.." && pwd)
 DEPS_DIR="$BASE_DIR/deps"
 
-CURRENT_VERSION=$(grep "#define ZLIB_VERSION" "$DEPS_DIR/zlib/zlib.h" | sed -n "s/^.*VERSION \"\(.*\)\"/\1/p")
+echo "Comparing latest upstream with current revision"
 
-NEW_VERSION_ZLIB_H=$(curl -s "https://chromium.googlesource.com/chromium/src/+/refs/heads/main/third_party/zlib/zlib.h?format=TEXT" | base64 --decode)
+# Revert zconf.h changes before checking diff 
+perl -i -pe 's|^//#include "chromeconf.h"|#include "chromeconf.h"|' "$DEPS_DIR/zlib/zconf.h"
+git stash -- "$DEPS_DIR/zlib/zconf.h"
 
-NEW_VERSION=$(printf '%s' "$NEW_VERSION_ZLIB_H" | grep "#define ZLIB_VERSION" | sed -n "s/^.*VERSION \"\(.*\)\"/\1/p")
+git fetch https://chromium.googlesource.com/chromium/src/third_party/zlib.git HEAD
 
-echo "Comparing $NEW_VERSION with $CURRENT_VERSION"
+DIFF_TREE=$(git diff --diff-filter=d 'stash@{0}:deps/zlib' FETCH_HEAD)
 
-if [ "$NEW_VERSION" = "$CURRENT_VERSION" ]; then
+git stash drop
+
+if [ -z "$DIFF_TREE" ]; then
   echo "Skipped because zlib is on the latest version."
   exit 0
 fi
 
+# This is a rather arbitrary restriction. This script is assumed to run on
+# Sunday, shortly after midnight UTC. This check thus prevents pulling in the
+# most recent commits if any changes were made on Friday or Saturday (UTC).
+# Because of Google's own "Live at Head" philosophy, new bugs that are likely to
+# affect Node.js tend to be fixed quickly, so we don't want to pull in a commit
+# that was just pushed, and instead rather wait for the next week's update. If
+# no commits have been pushed in the last two days, we assume that the most
+# recent commit is stable enough to be pulled in.
```

Comment:
I wrote this comment specifically for GoogleTest. Are you sure that this exact wording, in its entirety, is true for the Chromium fork of zlib, too?

Reply:
@tniessen I left your comment because your english is better than mine and is very well explained what happens in the next lines of code.

However I think that the text thus modified is correct; let me know if that's okay so I can update the comment. Thanks

> This is a rather arbitrary restriction. This script is assumed to run on
> Sunday, shortly after midnight UTC. This check thus prevents pulling in the
> most recent commits if any changes were made on Friday or Saturday (UTC).
> ~~Because of Google's own "Live at Head" philosophy, new bugs that are likely to
> affect Node.js tend to be fixed quickly, so~~ we don't want to pull in a commit
> that was just pushed, and instead rather wait for the next week's update. If
> no commits have been pushed in the last two days, we assume that the most
> recent commit is stable enough to be pulled in.